StreamDeckPlugin/Actions/CardButtonAction.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using ArkhamOverlay.TcpUtils;
+using ArkhamOverlay.TcpUtils.Requests;
+using ArkhamOverlay.TcpUtils.Responses;
+using Newtonsoft.Json.Linq;
+using SharpDeck;
+using SharpDeck.Events.Received;
+using SharpDeck.Manifest;
+using StreamDeckPlugin.Utils;
+
+namespace ArkhamOverlaySdPlugin.Actions {
+    public class CardButtonSettings {
+        public string Deck { get; set; }
+    }
+
+    [StreamDeckAction("Card Button", "arkhamoverlay.cardbutton")]
+    public class CardButtonAction : StreamDeckAction<CardButtonSettings> {
+        protected override Task OnSendToPlugin(ActionEventArgs<JObject> args) {
+            var deck = args.Payload["deck"].Value<string>();
+            var settings = new CardButtonSettings {
+                Deck = deck
+            };
+
+            SetSettingsAsync(settings);
+
+
+            var coordinates = new Coordinates {
+                Column = args.Payload["coordinates"]["column"].Value<int>(),
+                Row = args.Payload["coordinates"]["row"].Value<int>()
+            };
+            GetButtonInfo(settings, coordinates);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task OnWillAppear(ActionEventArgs<AppearancePayload> args) {
+            var settings = args.Payload.GetSettings<CardButtonSettings>();
+            GetButtonInfo(settings, args.Payload.Coordinates);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task OnKeyDown(ActionEventArgs<KeyPayload> args) {
+            var settings = args.Payload.GetSettings<CardButtonSettings>();
+            var cardIndex = GetCardButtonIndex(args.Payload.Coordinates);
+            try {
+                var request = new ClickCardButtonRequest { Deck = settings.Deck.AsDeck(), Index = cardIndex };
+                var response = SendSocketService.SendRequest<CardInfoReponse>(request);
+
+                SetImageAsync(response.CardButtonType.AsImage());
+                return SetTitleAsync(TextUtils.WrapTitle(response.Name));
+            } catch {
+                return SetTitleAsync("");
+            }
+        }
+
+        private void GetButtonInfo(CardButtonSettings settings, Coordinates coordinates) {
```

Comment:
This method might need a rename, it seems to be setting the button info

Reply:
Method is retrieving info from the overlay app